src/main/java/Duke.java

```diff
@@ -1,10 +1,265 @@
+                    break;
+                }
+            }
+        } while(!isLastInput);
+        exit();
     }
+
+    private static String extractDescription(String userInput) throws DukeException {
+        int startIndex = userInput.indexOf(" ");
+        if (startIndex == -1) {
+            throw new DukeException();
+        }
+        int endIndex;
+        if (userInput.contains("/")) {
+            endIndex = userInput.indexOf("/");
+        } else {
+            endIndex = userInput.length();
+        }
+        String description = userInput.substring(startIndex+1, endIndex).trim();
+        if (description.equals("")) {
+            throw new DukeException();
+        }
+        return description;
+    }
+
+    private static boolean checkTaskListSize() {
+        if (taskCount == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private static void checkEmptyInput(String userInput) throws DukeException {
+        if (userInput.length() == 0) {
+            throw new DukeException();
+        }
+    }
+
+    private static void checkTaskNumberValidity(int taskNumber) throws DukeException {
+        if (taskNumber > taskCount || taskNumber <= 0) {
+            throw new DukeException();
+        }
+    }
+
+    private static void checkInputValidity(String taskType) throws DukeException {
+        switch (taskType) {
+        case "todo":
+            // Fallthrough
+        case "deadline":
+            // Fallthrough
+        case "event":
+            // Fallthrough
+            break;
+        default:
+            throw new DukeException();
+        }
+    }
+
+    public static boolean detectLastInput(String input) {
+        return input.equals("bye");
+    }
+
+    public static boolean detectListRequest(String input) {
+        return input.equals("list");
+    }
+
+    public static boolean detectMarkRequest(String input) {
+        return input.startsWith("mark");
+    }
+
+    public static boolean detectUnmarkRequest(String input) {
+        return input.startsWith("unmark");
+    }
+
+    public static void greet() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void addTask(Task t){
+        tasks[taskCount] = t;
+        taskCount++;
+        System.out.println("____________________________________________________________");
+        System.out.println("Got it. I've added this task:");
+        System.out.println(t);
+    }
+
+    public static void printTaskCount() {
+        System.out.println("Now you have " + taskCount + " task(s) in the list.");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void list() {
```

Comment:
Consider renaming the method to make its purpose clearer. Perhaps "listTasks" would me more suitable?